lib/protobuf/field/base_field.rb

```diff
@@ -69,10 +69,12 @@ def default
       end
 
       def default_value
-        @default_value ||= case
-                           when optional? then typed_default_value
-                           when repeated? then ::Protobuf::Field::FieldArray.new(self).freeze
-                           when required? then nil
+        @default_value ||= if optional? || required?
+                             typed_default_value
+                           elsif repeated?
+                             ::Protobuf::Field::FieldArray.new(self).freeze
+                           else
+                             fail "something went very wrong"
```

Comment:
maybe something like "unable to produce default value" or something? very ominous message